osgar/bus.py

```diff
@@ -129,13 +129,16 @@ def __init__(self, log, inputs, outputs):
         self.buffer_queue = deque()
         self.max_delay = timedelta()
         self.max_delay_timestamp = timedelta()
+        self.finished = threading.Event()
 
     def register(self, *outputs):
         for o in outputs:
             # ignore :gz and :null modifiers
             assert o.split(':')[0] in self.outputs.values(), (o, self.outputs)
 
     def listen(self):
+        if self.finished.is_set():
```

Comment:
should not this be inside the `while True` cycle?

Reply:
Hmm. The slots strike again :worried: